language-plutus-core/src/Language/PlutusCore/TypeCheck.hs

```diff
@@ -87,7 +87,7 @@ checkKind config ann ty = runTypeCheckM config . checkKindM ann ty
 inferType
     :: (AsTypeError e ann, MonadError e m, MonadQuote m)
     => TypeCheckConfig -> Term TyName Name ann -> m (NormalizedType TyName ())
```

Comment:
In principle, we could remove `Quote` from here, but it's much better to do less than to do more. Since we're already monadic here, there is not much benefit (except less calls to `runQuote` at call sites), but if somebody changes something, we can start getting heisenbugs, so we need precise policies on where it's allowed to escape `Quote`.